### ISSUE TITLE
Feature: add locale format string to localeData or somewhere else

### ISSUE THREAD

**Issue by muuvmuuv** [iamkun/dayjs#894]
I want to set a datefields default format based on the language preference which is set on start but the only way to get the format string for the correct locale is seen below. I would like to have a valid function or constant that I can use. The right place for that would be in localeData or in `locale()` IMO but I don't even undestand why there are two `locale` and `$locale`?

```js
dayjs().$locale().formats.L
```

![Bildschirmfoto 2020-05-03 um 16 20 11](https://user-images.githubusercontent.com/11534760/80916636-46a4cd00-8d5a-11ea-8a31-cca63730ed01.png)

**Comment by iamkun** [comment]
Please avoid using private dev API starts with `$`, it may react different between releases.

**Comment by muuvmuuv** [comment]
I'm not talking about updating a locale, I want to get the format string of "L" programmatically. Like I do it with `$` in the example. I'm aware that this is used internally but is the only solution for now for me.

**Comment by iamkun** [comment]
try `dayjs.Ls`

**Comment by muuvmuuv** [comment]
Okay this does the job `dayjs.Ls[dayjs.locale()].formats.L` but IMO it is a bit of hacky. It would be simpler if there would be a `L` prop that holds the current locale formats. Just for easier access and readability. Anyway I think this isn't documented anywhere, isn't it?

**Comment by iamkun** [comment]
maybe LocaleData plugin is a better solution? How dose moment.js deal with this?

**Comment by muuvmuuv** [comment]
No, because a) the `formats.L` is already in the dayjs object and b) LocaleData  does not return it in the format I need: "DD.MM.YYYY" or "DD/MM/YYYY" depending on the locale.

My suggestion is to add a prop like `Ls` but that always holds current locale formats. I don't know how dayjs handeles `.format("L")` internally but somehow it must use the same approach I'm using now or `$locale().formats.L`.

I'll have a look at the source code.

**Comment by iamkun** [comment]
Yes, through `this.$locale()` in source code

**Comment by muuvmuuv** [comment]
Okay so `$locale()` is 1:1 what I am doing currently by your suggestion with `dayjs.Ls` and what I initially was looking for:

https://github.com/iamkun/dayjs/blob/dev/src/index.js#L363-L365

I suggest exposing this from internal dev API to external public API or like you did with Ls and EN here:

https://github.com/iamkun/dayjs/blob/dev/src/index.js#L414-L415

so:

```js
dayjs.en = Ls[L]
dayjs.Ls = Ls
dayjs.L = L // now I can access the current locale formats with dayjs.L
```

**Comment by iamkun** [comment]
I'll check momentjs to see if there is a better API name or method. dayjs.Ls is too tricky to list on the doc.

**Comment by muuvmuuv** [comment]
I would just name it `dayjs.Locales` and `dayjs.Locale`. Readable and simple.

**Comment by muuvmuuv** [comment]
Hey, did you came up with a naming?

I found the way momentjs do it:

```js
const m = require("moment")

const data = m.localeData()

console.log(data.longDateFormat("L"))
```

**Another** cool thing momentjs does is that when passing a locale like *de-DE* it tries to find it anyway:

> Moment will also try locale specifier substrings from most-specific to least-specific until it finds a locale it knows. This is
> useful when supplying Moment with a locale string pulled from the user's environment, such as window.navigator.language.
> 
> moment.locale('en-nz'); // 'en'
> Finally, Moment will search intelligently through an array of locales and their substrings.
> 
> moment.locale(['en-nz', 'en-au']); // 'en-au', not 'en'

**Comment by iamkun** [comment]
@muuvmuuv maybe we could extend our `localeData` plugin to do the same thing.

**Comment by muuvmuuv** [comment]
I don't see a reason why because this stuff is already in the dayjs object. Like I wrote above. Name it Locales  and Locale. Don't stick with the moment syntax when it is so simple.

**Comment by iamkun** [comment]
Moment like syntax make things easier to replace moment to dayjs

**Comment by muuvmuuv** [comment]
Ofc, it is just my opinion. At the end you have to decide.
